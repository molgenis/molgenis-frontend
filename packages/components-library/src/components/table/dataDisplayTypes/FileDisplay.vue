<template>
  <div class="text-nowrap text-truncate mouse-ref">
    <a href="details" @click.prevent="$bvModal.show(value.id)">
      {{ value.label }}
    </a>
    <a :href="value.url" class="ml-1">
      <font-awesome-icon icon="download" />
    </a>

    <b-modal :id="value.id" title="File details" hide-footer>
      <dl class="row">
        <dt class="col-sm-3">
          {{ $t('dataexplorer_file_modal_name_label') }}
        </dt>
        <dd class="col-sm-9">
          {{ value.filename }}
        </dd>
        <dt class="col-sm-3">
          {{ $t('dataexplorer_file_modal_content_type_label') }}
        </dt>
        <dd class="col-sm-9">
          {{ value.contentType }}
        </dd>
        <dt class="col-sm-3">
          {{ $t('dataexplorer_file_modal_size_label') }}
        </dt>
        <dd class="col-sm-9">
          {{ value.size }} byte
        </dd>
        <dt class="col-sm-3">
          {{ $t('dataexplorer_file_modal_url_label') }}
        </dt>
        <dd class="col-sm-9">
          <a :href="value.url">{{ value.url }}</a>
        </dd>
      </dl>
    </b-modal>
  </div>
</template>

<script>
export default {
  name: 'FileDisplay',
  props: {
    value: {
      /*
      *  id:string
      *  filename:string
      *  contentType:string
      *  label:string
      *  size:number
      *  url:string (full url)
      */
      type: Object,
      required: true
    }
  }
}
</script>

<style scoped>
dl dt,
dl dd {
  flex: null;
  max-width: 100%;
}
</style>
