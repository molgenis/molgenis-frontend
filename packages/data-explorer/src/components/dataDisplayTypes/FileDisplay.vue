<template>
  <div class="text-nowrap text-truncate mouse-ref">
    <a href="details" @click.prevent="$bvModal.show(value.id)" >
      {{value.label}}
    </a>
    <a :href="value.url" class="ml-1">
      <font-awesome-icon icon="download" />
    </a>

    <b-modal :id="value.id" title="File details" hide-footer>
      <dl class="row">
        <dt class="col-sm-3">Filename</dt>
        <dd class="col-sm-9">{{value.filename}}</dd>
        <dt class="col-sm-3">ContentType</dt>
        <dd class="col-sm-9">{{value.contentType}}</dd>
        <dt class="col-sm-3">Size</dt>
        <dd class="col-sm-9">{{value.size}}</dd>
        <dt class="col-sm-3">Url</dt>
        <dd class="col-sm-9"><a :href=value.url>{{value.url}}</a></dd>
      </dl>
    </b-modal>
  </div>
</template>

<script>
export default {
  name: 'FileDisplay',
  props: {
    value: {
      /*
      *  id:string
      *  filename:string
      *  contentType:string
      *  label:string
      *  size:number
      *  url:string (full url)
      */
      type: Object,
      required: true
    }
  }
}
</script>

<style scoped>
dl dt, dl dd
{
  flex: null;
  max-width: 100%;
}
</style>
