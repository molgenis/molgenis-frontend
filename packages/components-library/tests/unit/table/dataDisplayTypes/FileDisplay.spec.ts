import { createLocalVue, shallowMount } from '@vue/test-utils'
import FileDisplay from '@/components/table/dataDisplayTypes/FileDisplay.vue'

describe('FileDisplay.vue', () => {
  const localVue = createLocalVue()

  const stubs = ['b-modal', 'font-awesome-icon']
  const value = {
    id: 'id',
    filename: 'file-name',
    contentType: 'content-type',
    label: 'label',
    size: 1024,
    url: 'http://some/files/file-id'
  }

  it('should display the file and download link', () => {
    const wrapper = shallowMount(FileDisplay, { localVue, propsData: { value }, stubs, mocks: { $t: (msg: any) => msg } })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
