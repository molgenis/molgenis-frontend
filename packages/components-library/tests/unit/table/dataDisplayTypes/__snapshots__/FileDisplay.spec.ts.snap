// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileDisplay.vue should display the file and download link 1`] = `
<div class="text-nowrap text-truncate mouse-ref"><a href="details">
    label
  </a> <a href="http://some/files/file-id" class="ml-1">
    <font-awesome-icon-stub icon="download"></font-awesome-icon-stub>
  </a>
  <b-modal-stub id="id" title="File details" hide-footer="">
    <dl class="row">
      <dt class="col-sm-3">
        dataexplorer_file_modal_name_label
      </dt>
      <dd class="col-sm-9">
        file-name
      </dd>
      <dt class="col-sm-3">
        dataexplorer_file_modal_content_type_label
      </dt>
      <dd class="col-sm-9">
        content-type
      </dd>
      <dt class="col-sm-3">
        dataexplorer_file_modal_size_label
      </dt>
      <dd class="col-sm-9">
        1024 byte
      </dd>
      <dt class="col-sm-3">
        dataexplorer_file_modal_url_label
      </dt>
      <dd class="col-sm-9"><a href="http://some/files/file-id">http://some/files/file-id</a></dd>
    </dl>
  </b-modal-stub>
</div>
`;
